website/blog/2022-03-04-modifying-blocks.md

```diff
@@ -0,0 +1,275 @@
+---
+title: Modifying blocks - Color Theme
+description: Step-by-step guide how to modify and expand functionality of existing blocks.
+slug: modifying-blocks-color-theme
+authors: iobrado
+date: 2022-03-21
+tags: [eightshift, boilerplate, components, blocks]
+hide_table_of_contents: false
+---
+
+In our previous post, we covered how to use Storybook and WP-CLI for adding additional blocks into your project. This post will walk you step-by-step on how to modify an existing block.
+<!--truncate-->
+
+Since Eightshift Development kit is a starter theme, made for developers to jumpstart and speed up their development, you're welcome to modify files directly in it. There is no need to create a child theme to protect it from updates.
+
+### Modifying a block or a component?
+Since we used Quote block in our previous post, we will continue using it as an example since it's fairly simple. One of the first questions you may ask yourself could be: "Should I modify a component or a block?". And the answer is - it depends.
+
+If you compare **_components/quote/quote.php_** (component) and **_custom/quote/quote.php_** (block), you'll notice most HTML code is inside a component, while block pretty much only renders the Quote component inside a wrapper. This example will walk you through the whole process of adding a new attribute and its options to a block.
+
+### What will we do?
+
+We want to add one new feature to this block. An option to choose a color theme for the Quote block. These are the specs:
+- background with rounded corners
+- three color themes: blue, green, yellow
+- the background has a lighter shade of the selected color
+- quote icon has a stronger shade of the selected color
+
+### Adding background
+
+Initial background styling is fairly straightforward. Navigate to **_src/Blocks/components/quote/quote-style.scss_** and paste the following code inside `.quote` class:
+```css
+padding: calc(var(--base-font-size) * 1.6rem);
+border-radius: calc(var(--base-font-size) * 2rem);
+background-color: global-settings(colors, light);
+```
+You may notice we're using `calc` instead of directly writing values in rems. This way makes it much easier to calculate pixel size since `calc(var(--base-font-size) * 1.6rem)` equals `16px`.
```

Comment:
You should mention that the `--base-font-size` is also set to a certain value.

Also, IIRC, @volfkarlo mentioned that we should abandon this because it can affect the performance?

Reply:
We can update this later after the fix is added to the libs